hw/ip/hmac/doc/programmers_guide.md

```diff
@@ -1,8 +1,9 @@
 # Programmer's Guide
 
-This chapter shows how to use the HMAC-SHA-2 IP by showing some snippets such as initialization, initiating SHA-2 or HMAC process and processing the interrupts.
+This chapter shows how to use the HMAC/SHA-2 IP by showing some snippets such as initialization, initiating SHA-2 or HMAC process and processing the interrupts.
 This code is not compilable but serves to demonstrate the IO required.
-More detailed and complete code can be found in the software under `sw/`, [ROM code](https://github.com/lowRISC/opentitan/blob/master/sw/device/silicon_creator/lib/drivers/hmac.c) and [HMAC DIF](https://github.com/lowRISC/opentitan/blob/master/sw/device/lib/dif/dif_hmac.c).
+More detailed and complete code can be found in the software under `sw/`, [ROM code](https://github.com/lowRISC/opentitan/blob/master/sw/device/silicon_creator/lib/drivers/hmac.c) and [cryptolib code](https://github.com/lowRISC/opentitan/blob/master/sw/device/lib/crypto/drivers/hmac.c).
```

Comment:
I would actually drop the reference to the silicon creator driver here and only link the regular cryptolib link. The implementation from silicon driver is raw, meaning that there are no checks to enforce correct streaming expectations (i.e. feeding message bytes in block granularity). If people start using those as baseline for further development, they are going to have to spend time trying to debug unnecessary issues (this actually happened).

Reply:
And I'd use relative links to files in the repository (e.g., `../../../sw/device/lib/crypto/drivers/hmac.c`) rather than links to the GitHub repo

Reply:
@ballifatih agree with your concern, but I  would keep the reference to the silicon creator just because it helps (me at least if I am onboarding into this) to know all the different instances in the project where the HW IP is being used by the SW, but I would explicitly mention that this is not the reference/baseline SW implementation, but to refer to the cryptolib for that.